server/gc/src/test/java/org/apache/accumulo/gc/GarbageCollectionTest.java

```diff
@@ -644,4 +768,47 @@ public void bulkImportReplicationRecordsPreventDeletion() throws Exception {
     assertEquals(1, gce.deletes.size());
     assertEquals("hdfs://foo.com:6000/accumulo/tables/2/t-00002/A000002.rf", gce.deletes.get(0));
   }
+
+  /**
+   * Minimal test to show that dir and prevRow are required for valid scan
+   */
+  @Test
+  public void testDirAndPrevRow() throws Exception {
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+
+    TestGCE gce = new TestGCE();
+    gce.candidates.add("/1636/default_tablet");
+    gce.addDirReference("1636", null, "/default_tablet");
+    gce.addPrevRowReference("1636", null);
+    gca.collect(gce);
+  }
+
+  /**
+   * Show that IllegalState is thrown when no dir entry present in metadata scan.
+   */
+  @Test
+  public void testNoDir() {
```

Comment:
Need to test two tablets for this, one that is not the last tablet and one that is the last tablet.  Not sure which case this test is testing.

Reply:
Maybe there should be a test for the last tablet? But being this is using the TestGCE - does it really need two tablets.  The metadata check is not following the chain - just looking that an entry exists.  The check will pass if any entry is seen - even if it was valid (say point to a non-existing tablet)

This seemed to be the minimum about of code to trigger the exception.  Is that not valid?

Reply:
There are two places the check happen, in the loop and after the loop.  Thinking it would be good to cover both of those checks in the tests.

Reply:
I will add that

Reply:
oh yeah, looking back my initial comment really fails to clearly communicate what I was thinking about code coverage.

Reply:
Added in 5a893dfbac